src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1955,15 +1952,7 @@ private void HandleCall(MethodDesc callee, MethodSignature signature, ILOpcode o
 
         private ExpressionEntry HandleCall(MethodDesc callee, MethodSignature signature, StackEntry[] argumentValues, ILOpcode opcode = ILOpcode.call, TypeDesc constrainedType = null, LLVMValueRef calliTarget = default(LLVMValueRef), TypeDesc forcedReturnType = null)
         {
-            if (opcode == ILOpcode.callvirt && callee.IsVirtual)
-            {
-                AddVirtualMethodReference(callee);
-            }
-            else if (callee != null)
-            {
-                AddMethodReference(callee);
-            }
-            var pointerSize = _compilation.NodeFactory.Target.PointerSize;
+            AddDependencyForMethodCall(callee, opcode);
```

Comment:
Do we need this one? It looks like this is the same pattern as the `HandleCall` below.

Reply:
Can't just remove it completely as it will then fail to add dependencies for `System.Type.GetFields` among others.  However moving it to `LLVMFunctionForMethod ` where these method calls pass through is in line with what you mention above, I hope, and seems better as that's where the other dependencies for method calls are now added.

Reply:
Yup, looks much better this way. Could you delete the now orphaned `AddDependencyForMethodCall`?

Reply:
Done, PR turning into a ctrl-x exercise and nothing wrong with that.